src/runtime/HalideRuntime.h

```diff
@@ -523,67 +525,157 @@ struct halide_type_t {
         : code(code), bits(bits), lanes(lanes) {
     }
 
+    halide_type_t() : code((halide_type_code_t)0), bits(0), lanes(0) {}
+
     /** Size in bytes for a single element, even if width is not 1, of this type. */
-    size_t bytes() { return (bits + 7) / 8; }
+    size_t bytes() const { return (bits + 7) / 8; }
+
+    /** Compare two types for equality */
+    bool operator==(const halide_type_t &other) const {
+        return (code == other.code &&
+                bits == other.bits &&
+                lanes == other.lanes);
+    }
+
+    bool operator!=(const halide_type_t &other) const {
+        return !(*this == other);
+    }
 #endif
 };
 
-#ifndef BUFFER_T_DEFINED
-#define BUFFER_T_DEFINED
+typedef struct halide_dimension_t {
```

Comment:
Does any of our code assume / require alignment or padding of this struct? (specifically, that it is exactly 16 bytes?)